R/utilities-sensitivity-analysis.R

```diff
@@ -170,36 +169,32 @@ runParallelSensitivityAnalysis <- function(simFilePath,
   totalNumberParameters <- length(variableParameterPaths)
 
   # Parallelizing among a total of min(numberOfCores,totalNumberParameters) cores
-
   # Create a vector, of length totalNumberParameters, consisting of a repeating
-  #sequence of integers from 1 to numberOfCores
+  # sequence of integers from 1 to numberOfCores
   seqVec <- (1 + ((1:totalNumberParameters) %% numberOfCores))
 
   # Sort seqVec to obtain an concatenated array of repeated integers,
-  #with the repeated integers ranging from from 1 to numberOfCores.
-  #These are the core numbers to which each parameter will be assigned.
+  # with the repeated integers ranging from from 1 to numberOfCores.
+  # These are the core numbers to which each parameter will be assigned.
   sortVec <- sort(seqVec)
 
   # Split the parameters of the model according to sortVec
   listSplitParameters <- split(x = variableParameterPaths, sortVec)
-
   tempLogFileNamePrefix <- file.path(defaultFileNames$workflowFolderPath(), "logDebug-core-sensitivity-analysis")
   tempLogFileNames <- paste0(tempLogFileNamePrefix, seq(1, numberOfCores))
+
+  # Generate a listcontaining names of SA CSV result files that will be output by each core
+  allResultsFileNames <- generateResultFileNames(numberOfCores = numberOfCores, folderName = resultsFileFolder, fileName = resultsFileName)
   logDebug(message = "Starting sending of parameters to cores")
   Rmpi::mpi.bcast.Robj2slave(obj = listSplitParameters)
   Rmpi::mpi.bcast.Robj2slave(obj = tempLogFileNamePrefix)
   Rmpi::mpi.bcast.Robj2slave(obj = individualParameters)

```

Comment:
Can you explain what this line does again when you have time?

Reply:
Which line number? 191?

Reply:
yes. In general. Robj2slave

Reply:
The cores don't have direct access to variables on master.  So `Robj2slave` is used to send R objects to the cores.  

`Rmpi::mpi.bcast.Robj2slave(obj = individualParameters)` is sending the parameters specific to the individual so that the simulation can be updated ahead of the SA.  

`Rmpi::mpi.bcast.Robj2slave(obj = listSplitParameters)` is sending the list of variation parameters that will be perturbed in the SA.

Reply:
sure. But what does that mean. Does that mean that a slave will have access to a variable called individualParameters?

I am assuming this only works with primitive types (string, boolean, numbers and list thereof e)?

Reply:
Yes to both :)

Reply:
ok cool. Thanks :)